test/protractor/pages/credential.js

```diff
@@ -25,11 +25,7 @@ api.markPublic = function() {
   expect(checkbox.isSelected()).to.eventually.be.false;
   checkbox.click();
   editModal.element(by.partialButtonText('Save')).click();
-  browser.wait(function() {
-    return editModal.isPresent().then(function(res) {
-      return !res;
-    });
-  }, 8000);
+  browser.wait(EC.stalenessOf(editModal), 8000);
```

Comment:
stalenessOf waits for the element to get detatched from the DOM, which is what we want here. invisibilityOf just waits for it to stop being displayed.